crates/chain-config/src/config/state.rs

```diff
@@ -44,9 +65,108 @@ impl StateConfig {
             coins: db.get_coin_config()?,
             contracts: db.get_contract_config()?,
             messages: db.get_message_config()?,
-            height: Some(db.get_block_height()?),
         })
     }
+
+    #[cfg(feature = "std")]
+    pub fn load_from_directory(path: &str) -> Result<Self, anyhow::Error> {
+        let path = PathBuf::from_str(path)?.join("chain_state.json");
+        Self::load_from_file(path)
+    }
+
+    #[cfg(feature = "std")]
+    pub fn load_from_file(path: impl AsRef<Path>) -> Result<Self, anyhow::Error> {
+        let contents = std::fs::read(path.as_ref())?;
+        serde_json::from_slice(&contents).map_err(|e| {
+            anyhow::Error::new(e).context(format!(
+                "an error occurred while loading the chain state file: {:?}",
+                path.as_ref().to_str()
+            ))
+        })
+    }
+
+    #[cfg(feature = "std")]
+    pub fn local_testnet() -> Self {
+        // endow some preset accounts with an initial balance
+        tracing::info!("Initial Accounts");
+        let secrets = [
+            "0xde97d8624a438121b86a1956544bd72ed68cd69f2c99555b08b1e8c51ffd511c",
```

Comment:
Less concerned about having these as constants, although it could be useful for testing if the secrets are easy to reference.